### PR TITLE
[5.1] Implement Attributable trait

### PR DESCRIPTION
**Pull request by rkgrep**
API extension - allows to use Fluent features without need to extend Fluent itself.

The class automaticaly become compatible with `ArrayAccess`, `Arrayable`, `Jsonable` and `JsonSerializable` but it can extend some other class wich was not.

And an example of use in GenericUser - which actualy was copying most of the methods provided in trait.

### CODE REVIEW

File: src/Illuminate/Support/Traits/Attributable.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	 * @return $this
+	 */
+	public function __call($method, $parameters)
+	{
+		$this->attributes[$method] = count($parameters) > 0 ? $parameters[0] : true;
+
+		return $this;
+	}
+
+}
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Traits/Attributable.php]: missing new line

**Comment by GrahamCampbell** [comment]
Interesting idea.

**Comment by taylorotwell** [comment]
Going to hold off on this for now.

**Comment by sebastiandedeyne** [comment]
I like this idea, would be nice to be able to pull this in in non-laravel projects

**Comment by GrahamCampbell** [comment]
> would be nice to be able to pull this in in non-laravel projects

Feel free to turn it into a package provided you discuss licensing with @rkgrep as this code is his intellectual property.

**Comment by rkgrep** [comment]
@sebastiandedeyne You can find it in my repo https://github.com/rkgrep/attributable

**Pull request closed without merging**